### PR TITLE
Add Node Security page

### PR DESCRIPTION
**Pull request by blukat29**
## Proposed changes

- Add the "Securing nodes" page under the Nodes > Core Cell guides.
- It's about general security advice for node operators.

Note to maintainers:

- I tried to build the docs but failed with a "Hook useDoc is called outside the <DocProvider>" error. Therefore I am only submitting a markdown file.
- I want to locate this page in the Core Cell section. Teach me how or configure yourself, please.
	```
	[Nodes]
	Core Cell
		System Requirements
		Network Configuration
		Install Core Cell
		Monitor Core Cell
		Configure High Availability
		Securing Nodes <-- here
	```

## Types of changes

Please put an x in the boxes related to your change.

- [ ] Minor Issues and Typos
- [x] Major Content Contribution
- [ ] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to reach out. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [x] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/klaytn-docs/blob/master/CONTRIBUTING.md)
- [x] I have signed the [CLA](https://cla-assistant.io/klaytn/klaytn-docs)
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Related issues

- Please leave the issue numbers or links related to this PR here.

## Further comments

If this is a relatively large content contribution, kick off the discussion by explaining why you would suggest the content contribution, etc...

### CODE REVIEW

**Comment by ayo-klaytn** [comment]
Hi @blukat29 
Kindly follow this guide to include the newly added content in the Nav Sidebar section of the docs

Screenshot 2024-04-29 at 23 51 39

Screenshot 2024-04-29 at 23 54 02

Having created the securing-nodes.md file, you need to add it to the sidebar.js file for it to be properly indexed on the docs Nav Sidebar.

I hope it helps 🤝

File: docs/nodes/core-cell/securing-nodes.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+## Node security
+
+Follow the security best practices whenever possible. To list a few:
+
+- Keep the operating systems and installed software up-to-date.
+- SSH security. Disable remote root login, disable password login, and prefer public-key authentication.
+- Nodes should be run on a dedicated machine. To reduce the risk of malware via supply chain attacks, refrain from installing other software, especially the ones that connect to the Internet. The machine or virtual machine must be used for node operation purposes only. 
+- Install firewalls. Keep the open ports minimal and whitelist the maintenance and operator IP addresses. Be careful when opening the SSH, RPC, or debug ports. For CNs and PNs, the p2p port should be restricted to an explicit list of peer CN and PN IPs. Even if the p2p stack has a vulnerability the attack opportunity is effectively blocked or restricted to the known nodes.
+- Cloud security. When the node is operating on cloud virtual machines, properly manage the cloud accounts (e.g. AWS IAM roles) according to the cloud provider's best practices. The cloud accounts may not have access to SSH, but they can copy the disk contents or open a serial console.
```
Comment (scott-klaytn) [inline docs/nodes/core-cell/securing-nodes.md:17]: Suggestion: "Cloud security." -> "Prioritize cloud security."

File: docs/nodes/core-cell/securing-nodes.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+## Key security
+
+There are two major cryptographic keys managed by a node.
+
+- node key or p2p key: A 32-byte secp256k1 ECDSA private key. It's used in the p2p rlpx communication as well as signing the blocks.
```
Comment (scott-klaytn) [inline docs/nodes/core-cell/securing-nodes.md:23]: "node key" -> "Node key"

File: docs/nodes/core-cell/securing-nodes.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+- BLS node key: A 32-byte BLS12-381 private key. It's used to sign the RANDAO field in the blocks.
+
+Because the signing must happen automatically unattended, those keys have to be stored on the disk or entered via the command line. At this point, node security is critical in protecting the keys from theft.
+
+In the future, the nodes could support storing the keys in an external provider such as key management systems (KMS) or hardware security modules (HSM). Note that node security is still important even with these key management mechanisms. The keys cannot be copied but a compromised node can command signing malicious payload.
```
Comment (scott-klaytn) [inline docs/nodes/core-cell/securing-nodes.md:28]: Suggestion: "The keys cannot be copied but a compromised node can command signing malicious payload." -> "Although the keys cannot be copied, a compromised node can still initiate the signing of malicious payloads."

**Comment by blukat29** [comment]
@ayo-klaytn Thank you. I'll do it.
@scott-klaytn I failed when I `rm -rf node_modules` and reinstalled. But I succeeded with `yarn upgrade`. Thanks.

**Comment by blukat29** [comment]
@ayo-klaytn Added to the sidebar. Working on my local environment. I changed the page title from "Securing nodes" to "Node Security".
@scott-klaytn Edited as per your comments.

**Review by ayo-klaytn — APPROVED** [review]

**Review by scott-klaytn — APPROVED** [review]

**Pull request merged**